packages/orchestration/src/exos/remote-chain-facade.js

```diff
@@ -145,18 +211,51 @@ const prepareRemoteChainFacadeKit = (
          * }} ctx
          */
         onFulfilled(childNode, { account, chainAddress }) {
+          const { remoteChainInfo, connectionInfo } = this.state;
+          const { icqEnabled } = remoteChainInfo;
+
+          const icqConnVowOrUndefined =
+            icqEnabled && !this.state.icqConnection
+              ? E(orchestration).provideICQConnection(
+                  connectionInfo.counterparty.connection_id,
+                )
+              : undefined;
```

Comment:
```suggestion
          const { remoteChainInfo, connectionInfo, icqConnection } = this.state;
          const { icqEnabled } = remoteChainInfo;

          const icqConnVowOrUndefined =
            icqConnection ||
            (icqEnabled
              ? E(orchestration).provideICQConnection(
                  connectionInfo.counterparty.connection_id,
                )
              : undefined);
```

Reply:
_Current logic doesn't handle the scenario where we have an `icqConnection` in state already_